lsix

```diff
@@ -230,20 +230,24 @@ main() {
     [[ "$fontsize" ]] &&     imoptions+=" -pointsize $fontsize "
 
     # Create and display montages one row at a time.
-    while [ $# -gt 0 ]; do
-	# While we still have images to process...
-	onerow=()
-	goal=$(($# - numtiles)) # How many tiles left after this row
-	while [ $# -gt 0  -a  $# -gt $goal ]; do
-	    len=${#onerow[@]}
-	    onerow[len++]="-label"
-	    onerow[len++]=$(processlabel "$1")
-	    onerow[len++]="file://$1"
-	    shift
-	done
-	montage "${onerow[@]}"  $imoptions gif:-  \
-	    | convert - -colors $numcolors sixel:-
-    done
+    # Create and display montages one row at a time.
+		while [ $# -gt 0 ]; do
+				# While we still have images to process...
+				onerow=()
+				goal=$(($# - numtiles)) # How many tiles left after this row
+				while [ $# -gt 0 ] && [ $# -gt $goal ]; do
+						len=${#onerow[@]}
+						onerow[len++]="-label"
+						onerow[len++]="$(processlabel "$1")"
+						onerow[len++]="file://$1"
+						shift
+				done
+
+				# Note: The 'montage' command is not affected by this change, so it remains the same.
+				montage "${onerow[@]}" $imoptions gif:- \
+						| magick - -colors $numcolors sixel:-  # Convert to magick here.
```

Comment:
This would fail on a system with ImageMagick 6 which does not have a Magick command.